### PR TITLE
Swift: Improve swift/string-length-conflation

### PR DESCRIPTION
**Pull request by geoffw0**
Improve `swift/string-length-conflation`:
 - catch an edge case we were missing where a method was declared in a base class of the type we had expected it to be in.
 - clean up the logic a bit.

### CODE REVIEW

File: swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql [hunk 1]
```diff
@@ -58,39 +58,41 @@ class StringLengthConflationConfiguration extends DataFlow::Configuration {
   override predicate isSource(DataFlow::Node node, string flowstate) {
     // result of a call to `String.count`
     exists(MemberRefExpr member |
-      member.getBase().getType().getName() = "String" and
+      member.getBase().getType().(NominalType).getABaseType*().getName() = "String" and
```
Comment (MathiasVP) [inline swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql]: Could you not use the predicate I added in https://github.com/github/codeql/pull/10254 and rewrite this to:
```ql
member.getBase().(ClassDecl).getABaseTypeDecl*().getName()
```
(or something along those lines)?

**Review by MathiasVP — COMMENTED** [review]
One small comment. It also looks like this PR includes commits already merged into `main`? (i.e., [this one](https://github.com/github/codeql/pull/10256) and [this one](https://github.com/github/codeql/pull/10254))

File: swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql (hunk 1 above)
Comment (geoffw0) [reply swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql]: `getBase()` returns an `Expr`, not a `ClassDecl`, so I think an extra step of some form is required either way.

Comment (MathiasVP) [reply swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql]: Doh, right 🤦 . Yeah, I see there are multiple ways of doing this, but none of them are _obviously_ shorter than the rest.

**Review by MathiasVP — APPROVED** [review]

**Comment by geoffw0** [comment]
> It also looks like this PR includes commits already merged into main?

Yes, I believe that commit was required for this change (or at least I thought it was at one point).  I was having trouble with latest `main` so I pulled it in specifically instead.  It should come out in the merge.

**Pull request merged**